### DISCUSSION TITLE
Is Ollama required to run Open WebUI at `0.1.115`?

### DISCUSSION THREAD

**Discussion by changchiyou** [open-webui/open-webui#1287] [Q&A] [answered]
According to [[0.1.103] - 2024-02-25](https://github.com/open-webui/open-webui/blob/main/CHANGELOG.md#changed-8), it appears that Ollama is no longer a prerequisite for running Open WebUI since version `0.1.103`.

![image](https://github.com/open-webui/open-webui/assets/46549482/d1ce394f-008f-4817-8464-9d76523a0a20)

However, I encountered the following error messages when I run Open WebUI without Ollama:

```
open-webui         | INFO:apps.ollama.main:get_all_models()
open-webui         | ERROR:apps.ollama.main:Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
open-webui         | INFO:apps.ollama.main:get_all_models()
open-webui         | ERROR:apps.ollama.main:Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
```

Have I missed some documentation / settings? If yes, please let me know 😄 .

**Comment by changchiyou** [comment]
Any news?

**Reply by tjbck** [reply]
Ollama isn't required but it will still make requests by default, are you encountering any issues because of it?

**Reply by changchiyou** [reply]
Requesting the ollama endpoint might result in a white screen appearing ( https://github.com/open-webui/open-webui/discussions/1297#discussioncomment-8911449 ).

**Comment by rcontesti** [comment]
Hi, any news on this? I do not really use Ollama and it makes the installation heavier. Any way to remove it from a docker installation?

**Reply by riadbelaid** [reply, marked as answer]
> Hi, any news on this? I do not really use Ollama and it makes the installation heavier. Any way to remove it from a docker installation?

You can disable the Ollama requirement by setting [`ENABLE_OLLAMA_API=False`](https://docs.openwebui.com/getting-started/advanced-topics/env-configuration#enable_ollama_api), e.g.:

```bash
docker run -d -p 3000:8080 -e ENABLE_OLLAMA_API=False -v open-webui:/app/backend/data --name open-webui ghcr.io/open-webui/open-webui:main
```

**Reply by nirfse** [reply]
Can OWUI truly be run without the ollama dependency? When running with this ENVVAR I still can't use any (non-local, external API hosted) reasoning models, getting the following error in the logs after ~250 tokens of reasoning: `requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=11434): Max retries exceeded with url: /api/embed (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f64096cc090>: Failed to establish a new connection: [Errno 111] Connection refused'))`.
